driver/modern_bpf/helpers/store/auxmap_store_params.h

```diff
@@ -1568,43 +1577,43 @@ static __always_inline void apply_dynamic_snaplen(struct pt_regs *regs,
 	case PPME_SOCKET_SENDMSG_X: {
 		extract__network_args(args, 3, regs);
 		if(bpf_in_ia32_syscall()) {
-			struct compat_msghdr compat_mh = {};
-			if(likely(bpf_probe_read_user(&compat_mh,
+			if(likely(bpf_probe_read_user(&msg_mh.compat_mh,
 			                              bpf_core_type_size(struct compat_msghdr),
 			                              (void *)args[1]) == 0)) {
-				sockaddr = (struct sockaddr *)(unsigned long)(compat_mh.msg_name);
+				sockaddr = (struct sockaddr *)(unsigned long)(msg_mh.compat_mh.msg_name);
 			}
 			// in any case we break the switch.
 			break;
 		}
-
-		struct user_msghdr mh = {};
-		if(bpf_probe_read_user(&mh, bpf_core_type_size(struct user_msghdr), (void *)args[1]) == 0) {
-			sockaddr = (struct sockaddr *)mh.msg_name;
+		if(bpf_probe_read_user(&msg_mh.mh,
+		                       bpf_core_type_size(struct user_msghdr),
+		                       (void *)args[1]) == 0) {
+			sockaddr = (struct sockaddr *)msg_mh.mh.msg_name;
 		}
 	} break;
 
 	case PPME_SOCKET_RECVMMSG_X:
 	case PPME_SOCKET_SENDMMSG_X: {
-		extract__network_args(args, 3, regs);
+		// To avoid verifier stack size issues, sendmmsg and recvmmsg directly pass args
+		// in dynamic_snaplen_args.
+		// This also gives a small perf boost while using `bpf_loop` because we don't need
+		// to re-fetch first 3 syscall args at every iteration.
+		__builtin_memcpy(args, input_args->mm_args, 3 * sizeof(unsigned long));
```

Comment:
`sendmmsg/recvmmsg` improvement: since we are going to call `apply_dynamic_snaplen` for each `bpf_loop` iteration, just call `extract__network_args` once (in the `sendmmsg/recvmmsg` main program) and then reference it.
This also fixes a verifier issue on clang-14 about stack size too large.